### PR TITLE
Add ability to change log prefix depending on platform

### PR DESCRIPTION
**Pull request by imurchie**
Updating logging so the prefix is changed depending on the platform. This will not work if there are multiple simulators of different platforms (since the logger is shared) but that bridge can be crossed if/when it is reached.

Logs become like:
```
info tvOSSim Constructing tvOS simulator for Xcode version 9.4.1 with udid 'C2DEBF4F-D7D4-4A3F-8A31-13E86B505B1B'
info iOSSim Constructing iOS simulator for Xcode version 9.4.1 with udid '697D0C48-A7E6-4BBC-9112-015BE58D9432'
info watchOSSim Constructing watchOS simulator for Xcode version 9.4.1 with udid '861897A6-9E93-4423-9D5E-8D9FCDD84611'
```

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ class SimulatorXcode6 extends EventEmitter {
    */
   constructor (udid, xcodeVersion) {
     super();
+
```
Comment (KazuCocoa) [inline lib/simulator-xcode-6.js]: new empty line?

File: lib/utils.js [hunk 2]
```diff
@@ -167,11 +167,15 @@ async function simExists (udid) {
   }).reduce((a, b) => {
     return a.concat(b);
```
Comment (mykola-mokhnach) [inline lib/utils.js]: these returns are redundant

Comment (mykola-mokhnach) [reply lib/utils.js]: reduce can be replaced with flatMap

Comment (imurchie) [reply lib/utils.js]: I'd rather not have to mess with polyfills to get functionality we can already get from existing functions.

Comment (mykola-mokhnach) [reply lib/utils.js]: I mean lodash's flatMap

**Pull request merged**